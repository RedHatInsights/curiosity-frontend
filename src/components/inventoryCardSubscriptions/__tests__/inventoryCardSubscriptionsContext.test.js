import {
  context,
  useGetSubscriptionsInventory,
  useOnPageSubscriptions,
  useOnColumnSortSubscriptions
} from '../inventoryCardSubscriptionsContext';
import {
  RHSM_API_QUERY_INVENTORY_SORT_DIRECTION_TYPES as SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_INVENTORY_SUBSCRIPTIONS_SORT_TYPES as SORT_TYPES
} from '../../../services/rhsm/rhsmConstants';

describe('InventoryCardSubscriptionsContext', () => {
  it('should return specific properties', () => {
    expect(context).toMatchSnapshot('specific properties');
  });

  it('should expect specific sort properties', () => {
    expect(SORT_TYPES).toMatchSnapshot('sort properties');
  });

  it('should handle a store response with useGetSubscriptionsInventory', async () => {
    const { result } = await renderHook(
      () =>
        useGetSubscriptionsInventory({
          getInventory: () => () => {},
          useProduct: () => ({ productId: 'lorem' }),
          useDispatch: () => {}
        }),
      {
        state: {
          inventory: {
            subscriptionsInventory: {
              lorem: {
                fulfilled: true,
                data: [{ data: [{ lorem: 'ipsum' }, { dolor: 'sit' }], meta: {} }]
              }
            }
          }
        }
      }
    );

    expect(result).toMatchSnapshot('store response');
  });

  it('should handle variations in instances inventory API responses', async () => {
    const { result: errorResponse } = await renderHook(() =>
      useGetSubscriptionsInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ error: true })
      })
    );

    expect(errorResponse).toMatchSnapshot('inventory, error');

    const { result: pendingResponse } = await renderHook(() =>
      useGetSubscriptionsInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ pending: true })
      })
    );

    expect(pendingResponse).toMatchSnapshot('inventory, pending');

    const { result: cancelledResponse } = await renderHook(() =>
      useGetSubscriptionsInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ cancelled: true })
      })
    );

    expect(cancelledResponse).toMatchSnapshot('inventory, cancelled');

    const { result: fulfilledResponse } = await renderHook(() =>
      useGetSubscriptionsInventory({
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ fulfilled: true })
      })
    );

    expect(fulfilledResponse).toMatchSnapshot('inventory, fulfilled');

    const { result: disabledResponse } = await renderHook(() =>
      useGetSubscriptionsInventory({
        isDisabled: true,
        getInventory: () => () => {},
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useSelectorsResponse: () => ({ fulfilled: true })
      })
    );

    expect(disabledResponse).toMatchSnapshot('inventory, disabled');
  });

  it('should handle an onPage event', () => {
    const mockDispatch = jest.fn();
    const onPage = useOnPageSubscriptions({
      useDispatch: () => mockDispatch,
      useProduct: () => ({ productId: 'lorem' })
    });

    onPage({ offset: 1, perPage: 5 });
    expect(mockDispatch.mock.calls).toMatchSnapshot('onPage event, dispatch');
    mockDispatch.mockClear();
  });

  it('should handle an onColumnSort event', () => {
    const mockDispatch = jest.fn();
    const onColumnSort = useOnColumnSortSubscriptions({
      sortColumns: { LOREM_IPSUM_COLUMN_ONE: 'loremIpsumColumnOne' },
      useDispatch: () => mockDispatch,
      useProduct: () => ({ productId: 'lorem' })
    });

    onColumnSort(null, { direction: SORT_DIRECTION_TYPES.DESCENDING, id: 'loremIpsumColumnOne' });
    onColumnSort(null, { direction: SORT_DIRECTION_TYPES.ASCENDING, id: 'loremIpsumColumnOne' });

    expect(mockDispatch.mock.calls).toMatchSnapshot('onColumnSort event, dispatch');
    mockDispatch.mockClear();
  });
});
