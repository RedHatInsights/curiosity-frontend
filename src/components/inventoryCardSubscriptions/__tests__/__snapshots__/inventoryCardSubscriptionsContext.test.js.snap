// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryCardSubscriptionsContext should expect specific sort properties: sort properties 1`] = `
{
  "NEXT_EVENT_DATE": "next_event_date",
  "NEXT_EVENT_TYPE": "next_event_type",
  "PRODUCT_NAME": "product_name",
  "QUANTITY": "quantity",
  "SERVICE_LEVEL": "service_level",
  "SKU": "sku",
  "TOTAL_CAPACITY": "total_capacity",
  "USAGE": "usage",
}
`;

exports[`InventoryCardSubscriptionsContext should handle an onColumnSort event: onColumnSort event, dispatch 1`] = `
[
  [
    [
      {
        "dir": "desc",
        "type": "SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_dir",
        "viewId": "lorem",
      },
      {
        "sort": "loremIpsumColumnOne",
        "type": "SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
  [
    [
      {
        "dir": "asc",
        "type": "SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_dir",
        "viewId": "lorem",
      },
      {
        "sort": "loremIpsumColumnOne",
        "type": "SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_sort",
        "viewId": "lorem",
      },
    ],
  ],
]
`;

exports[`InventoryCardSubscriptionsContext should handle an onPage event: onPage event, dispatch 1`] = `
[
  [
    [
      {
        "offset": 1,
        "type": "SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_offset",
        "viewId": "lorem",
      },
      {
        "limit": 5,
        "type": "SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_limit",
        "viewId": "lorem",
      },
    ],
  ],
]
`;

exports[`InventoryCardSubscriptionsContext should return specific properties: specific properties 1`] = `
{
  "useGetSubscriptionsInventory": [Function],
  "useInventoryCardActionsSubscriptions": [Function],
  "useOnColumnSortSubscriptions": [Function],
  "useOnPageSubscriptions": [Function],
  "useParseSubscriptionsFiltersSettings": [Function],
  "useSelectorSubscriptions": [Function],
}
`;
